### PR TITLE
Notification channels + cache + collaboration provider

### PR DESCRIPTION
**Pull request by avrahams**

### CODE REVIEW

File: armotypes/cachetypes.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	val, ok := cache.Data.(T)
+	if !ok {
+		return val, fmt.Errorf("cache data is not of type %T", val)
+	}
+	return val, nil
+}
+
+// Generates cache data-type form properties
+func MakeCacheDataTypeV1(service, customerGUID, domain, propose, version string) DataType {
+	return DataType(fmt.Sprintf("datatypeV1:%s-%s-%s-%s-%s", service, customerGUID, domain, propose, version))
```
Comment (Bezbran) [inline armotypes/cachetypes.go:44]: *purpose
customerGUID should on the primary doc

File: armotypes/collaborationconfig.go [hunk 2]
```diff
@@ -44,13 +44,22 @@ type CollaborationConfigOption struct {
...
+)
+
 // swagger:model CollaborationConfig
 type CollaborationConfig struct {
 	PortalBase `json:",inline" bson:",inline"`
 
 	// Provider name
 	// Example: jira
-	Provider string `json:"provider" bson:"provider,omitempty"`
+	Provider ChannelProvider `json:"provider" bson:"provider,omitempty"`
```
Comment (Bezbran) [inline armotypes/collaborationconfig.go:62]: comment to exlain why both can be omitempty or not...

File: armotypes/usernotificationreporttypes.go [hunk 3]
```diff
@@ -53,56 +52,68 @@ type NotificationsConfig struct {
...
 }
 
-type PushReport struct {
-	Cluster                   string             `json:"custer,omitempty" bson:"custer,omitempty"`
-	ReportGUID                string             `json:"reportGUID,omitempty" bson:"reportGUID,omitempty"`
-	ScanType                  ScanType           `json:"scanType" bson:"scanType"`
-	Timestamp                 time.Time          `json:"timestamp,omitempty" bson:"timestamp,omitempty"`
-	FailedResources           uint64             `json:"failedResources,omitempty" bson:"failedResources,omitempty"`
-	FrameworksComplianceScore map[string]float32 `json:"frameworksComplianceScore,omitempty" bson:"frameworksComplianceScore,omitempty"`
+type NotificationParams map[NotificationParameter]interface{}
```
Comment (Bezbran) [inline armotypes/usernotificationreporttypes.go:67]: Please use struct, so we will have to make a PR and CR for changes of params

**Review by Bezbran — CHANGES_REQUESTED** [review]

File: armotypes/cachetypes.go (hunk 1 above)
Comment (avrahams) [reply armotypes/cachetypes.go:44]: Add omit empty to the json too

Comment (avrahams) [reply armotypes/cachetypes.go:44]: removed customer guid

File: armotypes/collaborationconfig.go (hunk 2 above)
Comment (avrahams) [reply armotypes/collaborationconfig.go:62]: add omit empty to json too

File: armotypes/usernotificationreporttypes.go [hunk 4]
```diff
@@ -129,3 +140,12 @@ type SeverityDetails struct {
 	Severity              string `json:"severity" bson:"severity"`
 	FailedResourcesNumber int    `json:"failedResourcesNumber" bson:"failedResourcesNumber"`
 }
+
+type NotificationPushEvent struct {
+	CustomerGUID string           `json:"customerGUID"`
+	EventName    string           `json:"eventName"`
+	EventTime    time.Time        `json:"eventTime"`
+	ReportGUID   string           `json:"reportGUID,omitempty"`
```
[the remark this answers is not in the document]
Comment (avrahams) [reply armotypes/usernotificationreporttypes.go]: removed cluster,reportguid and customer GUID - all got into designators

**Review by Bezbran — APPROVED** [review]

**Pull request merged**